udp/udp_rio_windows.go

```diff
@@ -323,6 +329,14 @@ func (u *RIOConn) Close() error {
 	windows.PostQueuedCompletionStatus(u.rx.iocp, 0, 0, nil)
 	windows.PostQueuedCompletionStatus(u.tx.iocp, 0, 0, nil)
 
+	u.rx.mu.Lock()                   // for waiting till active reader is done
```

Comment:
Had to run the test 1000 times to achieve stable reproduction on my windows 11. Is fixed with this change.